server/data/data-library/src/test/java/org/apache/james/user/lib/model/DefaultUserTest.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.james.user.lib.model;
+
+import org.apache.james.core.Username;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DefaultUserTest {
+
+    private DefaultUser user;
+
+    @Before
+    public void setUp() {
+        user = new DefaultUser(
+                Username.of("joe"),
+                "5en6G6MezRroT3XKqkdPOmY/", // SHA-1 legacy hash of "secret"
+                Algorithm.of("SHA-1", "legacy"),
+                Algorithm.of("SHA-256", "plain"));
+    }
+
+    @Test
+    public void shouldYieldVerifyAlgorithm() {
+        assertThat(user.getHashAlgorithm().asString()).isEqualTo("SHA-1");
+        assertThat(user.getHashAlgorithm().getHashingMode()).isEqualTo("LEGACY");
+    }
+
+    @Test
+    public void shouldVerifyPasswordUsingVerifyAlgorithm() {
+        assertThat(user.verifyPassword("secret")).isTrue();
+    }
+
+    @Test
+    public void shouldSetPasswordUsingUpdateAlgorithm() {
+        user.setPassword("secret2");
+        assertThat(user.getHashAlgorithm().asString()).isEqualTo("SHA-256");
+        assertThat(user.getHashAlgorithm().getHashingMode()).isEqualTo("PLAIN");
+
+        assertThat(user.verifyPassword("secret2")).isTrue();
```

Comment:
+ `assertThat(user.verifyPassword("secret")).isFalse();` 
// not really necessary

Reply:
You mean as a cross-check? Sure, will add it.